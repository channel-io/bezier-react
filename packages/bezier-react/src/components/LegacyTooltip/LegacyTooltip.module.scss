.LegacyTooltip {
  position: relative;
}

.LegacyTooltipContentWrapper {
  position: absolute;
  z-index: var(--z-index-tooltip);
  top: 0;
  left: 0;

  &:where(.disabled) {
    display: none;
  }
}

.LegacyTooltipContent {
  box-sizing: border-box;
  width: max-content;
  max-width: 260px;
  height: max-content;
  padding: 8px 14px;

  color: var(--txt-black-darkest);
  word-break: normal;
  word-wrap: break-word;

  background-color: var(--bg-white-low);
  border-radius: var(--radius-8);
  box-shadow: var(--ev-2);
}
