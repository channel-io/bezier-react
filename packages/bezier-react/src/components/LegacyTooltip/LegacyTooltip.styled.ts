import {
  LineHeightAbsoluteNumber,
  css,
  ellipsis,
  styled,
} from '~/src/foundation'

import { ZIndex } from '~/src/constants/ZIndex'
import type { InterpolationProps } from '~/src/types/Foundation'

interface ContentWrapperProps {
  disabled: boolean
}

export const Container = styled.div`
  position: relative;
`
export const ContentWrapper = styled.div<ContentWrapperProps>`
  position: absolute;
  top: 0;
  left: 0;
  z-index: ${ZIndex.Tooltip};

  ${({ disabled }) => disabled && css`
    display: none;
  `}
`

export const Content = styled.div<InterpolationProps>`
  box-sizing: border-box;
  width: max-content;
  max-width: 260px;
  height: max-content;
  padding: 8px 14px;
  color: ${({ foundation }) => foundation?.subTheme?.['txt-black-darkest']};
  word-break: normal;
  word-wrap: break-word;
  ${({ foundation }) => foundation?.elevation?.ev2(true)};
  ${({ foundation }) => foundation?.rounding?.round8};

  ${({ interpolation }) => interpolation}
`

export const EllipsisableContent = styled.div`
  ${ellipsis(20, LineHeightAbsoluteNumber.Lh18)}
`
