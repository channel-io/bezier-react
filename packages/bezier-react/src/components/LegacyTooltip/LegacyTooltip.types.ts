import { type Ref } from 'react'

import {
  type AdditionalOverridableStyleProps,
  type BezierComponentProps,
  type ChildrenProps,
  type ContentProps,
  type DisableProps,
  type PolymorphicProps,
} from '~/src/types/props'

interface LegacyTooltipOptions {
  placement?: LegacyTooltipPosition
  offset?: number
  keepInContainer?: boolean
  allowHover?: boolean
  delayShow?: number
  delayHide?: number
  contentTestId?: string
  lazy?: boolean
}

export interface LegacyTooltipProps extends
  Omit<BezierComponentProps<'div'>, keyof ContentProps>,
  PolymorphicProps,
  ChildrenProps,
  ContentProps,
  DisableProps,
  AdditionalOverridableStyleProps<'content' | 'contentWrapper'>,
  LegacyTooltipOptions {
}

export interface LegacyTooltipContentProps extends Pick<
LegacyTooltipOptions,
'keepInContainer' |
'placement' |
'offset' |
'allowHover'
>,
  Omit<BezierComponentProps<'div'>, keyof ContentProps>,
  PolymorphicProps,
  ContentProps,
  AdditionalOverridableStyleProps<'content' | 'contentWrapper'>,
  DisableProps {
  tooltipContainer: HTMLDivElement | null
  forwardedRef: Ref<HTMLDivElement>
}

export interface GetTooltipStyle extends Required<Pick<LegacyTooltipOptions, 'placement' | 'offset' | 'allowHover'>> {
  tooltipContainer: HTMLDivElement
}

export interface GetReplacement extends Required<Pick<LegacyTooltipOptions, 'placement' | 'keepInContainer'>> {
  tooltip: HTMLDivElement
  rootElement: HTMLElement
}

export enum LegacyTooltipPosition {
  TopCenter = 'topCenter',
  TopLeft = 'topLeft',
  TopRight = 'topRight',
  RightCenter = 'rightCenter',
  RightTop = 'rightTop',
  RightBottom = 'rightBottom',
  BottomCenter = 'bottomCenter',
  BottomLeft = 'bottomLeft',
  BottomRight = 'bottomRight',
  LeftCenter = 'leftCenter',
  LeftTop = 'leftTop',
  LeftBottom = 'leftBottom',
}
