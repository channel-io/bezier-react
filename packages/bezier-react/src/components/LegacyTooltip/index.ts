export { LegacyTooltip } from './LegacyTooltip'
export {
  type LegacyTooltipContentProps,
  LegacyTooltipPosition,
  type LegacyTooltipProps,
} from './LegacyTooltip.types'
