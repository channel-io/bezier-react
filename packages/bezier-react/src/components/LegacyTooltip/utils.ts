import {
  type GetReplacement,
  type GetTooltipStyle,
  LegacyTooltipPosition,
} from './LegacyTooltip.types'

export function getReplacement({
  tooltip,
  keepInContainer,
  placement,
  rootElement,
}: GetReplacement): LegacyTooltipPosition {
  if (!keepInContainer) {
    return placement
  }

  const {
    width: tooltipWidth,
    height: tooltipHeight,
    top: tooltipTop,
    left: tooltipLeft,
  } = tooltip.getBoundingClientRect()

  const {
    width: rootWidth,
    height: rootHeight,
    top: rootTop,
    left: rootLeft,
  } = rootElement.getBoundingClientRect()

  const isOverTop = tooltipTop < rootTop
  const isOverBottom = tooltipTop + tooltipHeight > rootTop + rootHeight
  const isOverLeft = tooltipLeft < rootLeft
  const isOverRight = tooltipLeft + tooltipWidth > rootLeft + rootWidth

  if (isOverTop) {
    if (isOverLeft) {
      return LegacyTooltipPosition.BottomLeft
    }
    if (isOverRight) {
      return LegacyTooltipPosition.BottomRight
    }
    return LegacyTooltipPosition.BottomCenter
  }

  if (isOverBottom) {
    if (isOverLeft) {
      return LegacyTooltipPosition.TopLeft
    }
    if (isOverRight) {
      return LegacyTooltipPosition.TopRight
    }
    return LegacyTooltipPosition.TopCenter
  }

  if (isOverLeft) {
    if (isOverTop) {
      return LegacyTooltipPosition.RightTop
    }
    if (isOverBottom) {
      return LegacyTooltipPosition.RightBottom
    }
    return LegacyTooltipPosition.RightCenter
  }

  if (isOverRight) {
    if (isOverTop) {
      return LegacyTooltipPosition.LeftTop
    }
    if (isOverBottom) {
      return LegacyTooltipPosition.LeftBottom
    }
    return LegacyTooltipPosition.LeftCenter
  }

  return placement
}

export function getTooltipStyle({
  tooltipContainer,
  placement,
  offset,
  allowHover,
}: GetTooltipStyle) {
  const {
    top: containerTop,
    left: containerLeft,
    width: containerWidth,
    height: containerHeight,
  } = tooltipContainer.getBoundingClientRect()

  let top = containerTop
  let left = containerLeft
  let translateX = 0
  let translateY = 0

  let offsetTop = 0
  let offsetRight = 0
  let offsetBottom = 0
  let offsetLeft = 0
  let offsetStyle = {}

  switch (placement) {
    case LegacyTooltipPosition.TopCenter:
    case LegacyTooltipPosition.TopLeft:
    case LegacyTooltipPosition.TopRight:
      translateY = -100
      offsetBottom = offset
      break
    case LegacyTooltipPosition.RightCenter:
    case LegacyTooltipPosition.RightTop:
    case LegacyTooltipPosition.RightBottom:
      left += containerWidth
      offsetLeft = offset
      break
    case LegacyTooltipPosition.BottomCenter:
    case LegacyTooltipPosition.BottomLeft:
    case LegacyTooltipPosition.BottomRight:
      top += containerHeight
      offsetTop = offset
      break
    case LegacyTooltipPosition.LeftCenter:
    case LegacyTooltipPosition.LeftTop:
    case LegacyTooltipPosition.LeftBottom:
      translateX = -100
      offsetRight = offset
      break
  }

  switch (placement) {
    case LegacyTooltipPosition.TopCenter:
    case LegacyTooltipPosition.BottomCenter:
      translateX = -50
      left += (containerWidth / 2)
      break
    case LegacyTooltipPosition.TopRight:
    case LegacyTooltipPosition.BottomRight:
      translateX = -100
      left += containerWidth
      break
    case LegacyTooltipPosition.RightCenter:
    case LegacyTooltipPosition.LeftCenter:
      translateY = -50
      top += (containerHeight / 2)
      break
    case LegacyTooltipPosition.RightBottom:
    case LegacyTooltipPosition.LeftBottom:
      translateY = -100
      top += containerHeight
      break
  }

  if (allowHover) {
    offsetStyle = {
      paddingTop: offsetTop,
      paddingRight: offsetRight,
      paddingBottom: offsetBottom,
      paddingLeft: offsetLeft,
    }
  } else {
    top += (offsetTop - offsetBottom)
    left += (offsetLeft - offsetRight)
  }

  return {
    position: 'fixed',
    top,
    left,
    transform: `translate(${translateX}%, ${translateY}%)`,
    ...offsetStyle,
  }
}
