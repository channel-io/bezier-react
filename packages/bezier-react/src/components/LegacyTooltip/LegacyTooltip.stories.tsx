import React from 'react'

import {
  type Meta,
  type StoryFn,
} from '@storybook/react'

import { styled } from '~/src/foundation'

import { LegacyTooltip } from './LegacyTooltip'
import { type LegacyTooltipProps } from './LegacyTooltip.types'
import { LegacyTooltipPosition } from './LegacyTooltip.types'

const meta: Meta<typeof LegacyTooltip> = {
  component: LegacyTooltip,
  argTypes: {
    content: {
      control: {
        type: 'text',
      },
    },
    placement: {
      control: {
        type: 'radio',
      },
      options: [
        LegacyTooltipPosition.TopCenter,
        LegacyTooltipPosition.TopLeft,
        LegacyTooltipPosition.TopRight,
        LegacyTooltipPosition.RightCenter,
        LegacyTooltipPosition.RightTop,
        LegacyTooltipPosition.RightBottom,
        LegacyTooltipPosition.BottomCenter,
        LegacyTooltipPosition.BottomLeft,
        LegacyTooltipPosition.BottomRight,
        LegacyTooltipPosition.LeftCenter,
        LegacyTooltipPosition.LeftTop,
        LegacyTooltipPosition.LeftBottom,
      ],
    },
    offset: {
      control: {
        type: 'range',
        min: 0,
        max: 50,
        step: 1,
      },
    },
    delayShow: {
      control: {
        type: 'range',
        min: 0,
        max: 5000,
        step: 100,
      },
    },
    delayHide: {
      control: {
        type: 'range',
        min: 0,
        max: 5000,
        step: 100,
      },
    },
  },
}
export default meta

const Target = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100px;
  height: 40px;
  background-color: ${props => props.foundation?.theme?.['bg-black-dark']};
  border-radius: 4px;
`

const Template: StoryFn<LegacyTooltipProps> = (props) => (
  <LegacyTooltip {...props}>
    <Target>Target</Target>
  </LegacyTooltip>
)

export const Primary = {
  render: Template,

  args: {
    // eslint-disable-next-line max-len
    content:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
    lazy: true,
    placement: LegacyTooltipPosition.BottomCenter,
    offset: 4,
    disabled: false,
    keepInContainer: false,
    allowHover: false,
    delayShow: 0,
    delayHide: 0,
  },
}
