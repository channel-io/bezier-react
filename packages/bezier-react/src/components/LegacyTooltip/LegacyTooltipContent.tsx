import React, {
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react'
import ReactDOM from 'react-dom'

import classNames from 'classnames'

import useEventHandler from '~/src/hooks/useEventHandler'
import useMergeRefs from '~/src/hooks/useMergeRefs'
import {
  isArray,
  isEmpty,
  isString,
} from '~/src/utils/type'

import { Text } from '~/src/components/Text'
import { InvertedThemeProvider } from '~/src/components/ThemeProvider'
import { useRootElement } from '~/src/components/WindowProvider'

import { type LegacyTooltipContentProps } from './LegacyTooltip.types'
import {
  getReplacement,
  getTooltipStyle,
} from './utils'

import styles from './LegacyTooltip.module.scss'

function getNewLineComponent(strContent: string) {
  return (
    strContent.split('\n').map((str, index) => {
      if (index === 0) {
        return (
          <Text key={str} typo="14">
            { str }
          </Text>
        )
      }

      return (
        <React.Fragment key={str}>
          <br />
          <Text typo="14">
            { str }
          </Text>
        </React.Fragment>
      )
    })
  )
}

function getContentComponent(content?: React.ReactNode) {
  if (isArray(content)) {
    return content.map(item => {
      if (isString(item)) {
        return getNewLineComponent(item)
      }

      return item
    })
  }

  if (isString(content)) {
    return getNewLineComponent(content)
  }

  return content
}

export const LegacyTooltipContent: React.FC<LegacyTooltipContentProps> = ({
  as,
  content,
  contentStyle,
  contentClassName,
  contentWrapperClassName,
  contentWrapperStyle: givenContentWrapperStyle,
  disabled = false,
  keepInContainer = false,
  placement = 'bottom-center',
  tooltipContainer,
  offset = 4,
  allowHover = false,
  forwardedRef,
  ...rest
}) => {
  const rootElement = useRootElement()

  const tooltipRef = useRef<HTMLDivElement>(null)
  const tooltipWrapperRef = useRef<HTMLDivElement>(null)
  const mergedRef = useMergeRefs<HTMLDivElement>(tooltipRef, forwardedRef)
  const [replacement, setReplacement] = useState(placement)

  const handleClickTooltip = useCallback((event: HTMLElementEventMap['click']) => {
    event.stopPropagation()
  }, [])

  useEventHandler(tooltipRef.current, 'click', handleClickTooltip)

  useEffect(() => {
    if (!tooltipRef.current) { return }
    const newPlacement = getReplacement({
      tooltip: tooltipRef.current,
      keepInContainer,
      placement,
      rootElement,
    })
    setReplacement(newPlacement)
  }, [
    rootElement,
    keepInContainer,
    placement,
  ])

  const ContentComponent = useMemo(() => getContentComponent(content), [content])

  const contentWrapperStyle = useMemo(() => {
    if (tooltipContainer) {
      return {
        ...givenContentWrapperStyle,
        ...getTooltipStyle({
          tooltipContainer,
          placement: replacement,
          offset,
          allowHover,
        }),
      }
    }

    return {}
  }, [
    tooltipContainer,
    givenContentWrapperStyle,
    replacement,
    offset,
    allowHover,
  ])

  const Comp = as ?? 'div'

  return (
    ReactDOM.createPortal(
      <InvertedThemeProvider>
        <div
          style={contentWrapperStyle}
          className={classNames(
            styles.LegacyTooltipContentWrapper,
            (disabled || isEmpty(content)) && styles.disabled,
            contentWrapperClassName,
          )}
          ref={tooltipWrapperRef}
        >
          <Comp
            style={contentStyle}
            className={classNames(
              styles.LegacyTooltipContent,
              contentClassName,
            )}
            ref={mergedRef}
            {...rest}
          >
            <Text
              color="txt-black-darkest"
              truncated={20}
              typo="13"
            >
              { ContentComponent }
            </Text>
          </Comp>
        </div>
      </InvertedThemeProvider>,
      rootElement,
    )
  )
}
