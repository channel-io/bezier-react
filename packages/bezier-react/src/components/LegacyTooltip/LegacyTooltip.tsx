import React, {
  type Ref,
  forwardRef,
  memo,
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react'

import classNames from 'classnames'

import { useWindow } from '~/src/components/WindowProvider'

import { type LegacyTooltipProps } from './LegacyTooltip.types'
import { LegacyTooltipContent } from './LegacyTooltipContent'

import styles from './LegacyTooltip.module.scss'

export const TOOLTIP_TEST_ID = 'bezier-tooltip'
export const TOOLTIP_CONTENT_TEST_ID = 'bezier-tooltip-content'

/**
 * @deprecated Use `Tooltip` instead. It may be removed in the next major version.
 */
export const LegacyTooltip = memo(
  forwardRef(function LegacyTooltip(
    {
      as,
      contentTestId = TOOLTIP_CONTENT_TEST_ID,
      className,
      contentStyle,
      contentClassName,
      contentWrapperStyle,
      contentWrapperClassName,
      content = null,
      lazy = false, // optional prop 에서 추후 default behavior 를 lazy 하게 바꿀 예정
      placement = 'bottom-center',
      disabled = false,
      offset = 4,
      keepInContainer = false,
      allowHover = false,
      delayShow = 0,
      delayHide = 0,
      children,
      ...otherProps
    }: LegacyTooltipProps,
    forwardedRef: Ref<any>
  ) {
    const { window } = useWindow()

    const [show, setShow] = useState(false)
    const [didMount, setDidMount] = useState(show)

    const tooltipContainerRef = useRef<HTMLDivElement>(null)
    const timerRef = useRef<ReturnType<Window['setTimeout']>>()

    useEffect(
      function hideTooltipContentWhenDisabled() {
        if (disabled) {
          setShow(false)
        }
      },
      [disabled]
    )

    useEffect(
      function updateDidMount() {
        setDidMount((prev) => prev || show)
      },
      [show]
    )

    const handleMouseEnter = useCallback(() => {
      if (disabled) {
        return
      }

      if (timerRef.current) {
        clearTimeout(timerRef.current)
      }

      timerRef.current = window.setTimeout(() => {
        setShow(true)
      }, delayShow)
    }, [delayShow, disabled, window])

    const handleMouseLeave = useCallback(() => {
      if (disabled) {
        return
      }

      if (timerRef.current) {
        clearTimeout(timerRef.current)
      }

      timerRef.current = window.setTimeout(() => {
        setShow(false)
      }, delayHide)
    }, [delayHide, disabled, window])

    const TooltipComponent = useMemo(() => {
      if (!lazy || didMount) {
        return (
          <LegacyTooltipContent
            as={as}
            content={content}
            contentStyle={contentStyle}
            contentClassName={contentClassName}
            contentWrapperStyle={contentWrapperStyle}
            contentWrapperClassName={contentWrapperClassName}
            disabled={disabled}
            placement={placement}
            offset={offset}
            allowHover={allowHover}
            keepInContainer={keepInContainer}
            tooltipContainer={tooltipContainerRef.current}
            forwardedRef={forwardedRef}
            data-testid={contentTestId}
          />
        )
      }

      return null
    }, [
      lazy,
      didMount,
      as,
      content,
      contentStyle,
      contentClassName,
      contentWrapperStyle,
      contentWrapperClassName,
      disabled,
      placement,
      offset,
      allowHover,
      keepInContainer,
      forwardedRef,
      contentTestId,
    ])

    if (!children) {
      return null
    }

    return (
      <div
        ref={tooltipContainerRef}
        className={classNames(styles.LegacyTooltip, className)}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        data-testid={TOOLTIP_TEST_ID}
        {...otherProps}
      >
        {children}
        {show && TooltipComponent}
      </div>
    )
  })
)
