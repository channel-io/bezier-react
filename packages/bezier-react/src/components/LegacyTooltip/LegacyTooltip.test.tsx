import React from 'react'

import { fireEvent } from '@testing-library/dom'
import { act } from '@testing-library/react'

import { css } from '~/src/foundation'

import { render } from '~/src/utils/test'

import {
  LegacyTooltip,
  TOOLTIP_CONTENT_TEST_ID,
  TOOLTIP_TEST_ID,
} from './LegacyTooltip'
import { type LegacyTooltipProps } from './LegacyTooltip.types'

const RootTooltip: React.FC<LegacyTooltipProps> = ({ children, ...rests }) => (
  <LegacyTooltip {...rests}>
    { children }
  </LegacyTooltip>
)

describe('Tooltip test >', () => {
  let props: LegacyTooltipProps

  const content = 'Hovered'

  beforeEach(() => {
    props = {
      children: 'Text',
      content,
    }

    jest.useFakeTimers()
  })

  afterEach(() => {
    jest.useRealTimers()
  })

  const renderTooltip = (optionProps?: LegacyTooltipProps) => render(
    <RootTooltip {...props} {...optionProps} />,
  )

  it('Tooltip with default props', () => {
    const { baseElement, getByTestId } = renderTooltip()

    act(() => {
      fireEvent.mouseOver(getByTestId(TOOLTIP_TEST_ID))

      jest.runAllTimers()
    })

    expect(baseElement).toMatchSnapshot()
  })

  it('Tooltip with contentInterpolation prop', async () => {
    const { baseElement, getByTestId } = renderTooltip({
      contentInterpolation: css`background-color: black;`,
    })

    act(() => {
      fireEvent.mouseOver(getByTestId(TOOLTIP_TEST_ID))

      jest.runAllTimers()
    })

    expect(baseElement).toMatchSnapshot()
  })

  it('TooltipContent not rendered at first', async () => {
    const { queryByTestId } = renderTooltip()

    expect(queryByTestId(TOOLTIP_CONTENT_TEST_ID)).toBeNull()
  })

  it('TooltipContent rendered after mouseover', async () => {
    const { getByTestId, queryByTestId } = renderTooltip()

    act(() => {
      fireEvent.mouseOver(getByTestId(TOOLTIP_TEST_ID))

      jest.runAllTimers()
    })

    expect(queryByTestId(TOOLTIP_CONTENT_TEST_ID)).toBeVisible()
  })

  it('Tooltip with disabled prop not rendered even after mouseover', async () => {
    const { getByTestId, queryByTestId } = renderTooltip({ disabled: true })

    act(() => {
      fireEvent.mouseOver(getByTestId(TOOLTIP_TEST_ID))

      jest.runAllTimers()
    })

    expect(queryByTestId(TOOLTIP_CONTENT_TEST_ID)).toBeNull()
  })
})
