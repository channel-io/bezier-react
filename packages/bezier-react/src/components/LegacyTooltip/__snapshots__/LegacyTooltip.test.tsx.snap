// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip test > Tooltip with contentInterpolation prop 1`] = `
.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1300;
}

.c2 {
  box-sizing: border-box;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 260px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  padding: 8px 14px;
  color: #FFFFFFCC;
  word-break: normal;
  word-wrap: break-word;
  background-color: #313234;
  box-shadow: inset 0 0 2px 0 #FFFFFF14, 0 0 2px 1px #00000014, 0 2px 6px #00000026;
  overflow: hidden;
  border-radius: 8px;
  background-color: black;
}

.c3 {
  display: -webkit-box;
  max-height: 36rem;
  overflow: hidden;
  line-height: 1.8rem;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 20;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="bezier-react-tooltip"
    >
      Text
    </div>
  </div>
  <div
    class="c1"
    style="position: fixed; top: 4px; left: 0px; transform: translate(-50%, 0%);"
  >
    <div
      class="c2"
      data-testid="bezier-react-tooltip-content"
    >
      <div
        class="c3"
      >
        <span
          class="Text typo-14 margin"
          data-testid="bezier-react-text"
        >
          Hovered
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`Tooltip test > Tooltip with default props 1`] = `
.c0 {
  position: relative;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1300;
}

.c2 {
  box-sizing: border-box;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 260px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  padding: 8px 14px;
  color: #FFFFFFCC;
  word-break: normal;
  word-wrap: break-word;
  background-color: #313234;
  box-shadow: inset 0 0 2px 0 #FFFFFF14, 0 0 2px 1px #00000014, 0 2px 6px #00000026;
  overflow: hidden;
  border-radius: 8px;
}

.c3 {
  display: -webkit-box;
  max-height: 36rem;
  overflow: hidden;
  line-height: 1.8rem;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 20;
}

<body>
  <div>
    <div
      class="c0"
      data-testid="bezier-react-tooltip"
    >
      Text
    </div>
  </div>
  <div
    class="c1"
    style="position: fixed; top: 4px; left: 0px; transform: translate(-50%, 0%);"
  >
    <div
      class="c2"
      data-testid="bezier-react-tooltip-content"
    >
      <div
        class="c3"
      >
        <span
          class="Text typo-14 margin"
          data-testid="bezier-react-text"
        >
          Hovered
        </span>
      </div>
    </div>
  </div>
</body>
`;
